File: frontend/lib/schemas/ai-addendum.ts
import { DocSchema } from "../doc-schema";

export const aiAddendumSchema: DocSchema = {
  id: "ai-addendum",
  title: "AI Addendum",
  subtitle: "Cover Page",
  infoBox:
    'This AI Addendum consists of: (1) this Cover Page and (2) the Common Paper AI Addendum Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/ai-addendum/1.0. The AI Addendum governs the use of artificial intelligence tools and services, including data use, model training restrictions, and AI-specific compliance terms.',
  coverFields: [
    {
      key: "effectiveDate",
      label: "Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "aiServices",
      label: "AI Services",
      hint: "Description of the AI services covered by this addendum",
      type: "textarea",
      placeholder: "Describe the AI services...",
    },
    {
      key: "dataUseRestrictions",
      label: "Data Use Restrictions",
      hint: "Restrictions on how customer data may be used with AI",
      type: "textarea",
      placeholder: "Describe any data use restrictions...",
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "chosenCourts",
      label: "Chosen Courts",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Provider", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Customer", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this AI Addendum as of the Effective Date.",
  templateFile: "templates/AI-Addendum.md",
  footerNote: "Common Paper AI Addendum (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your AI Addendum assistant. I'll help you fill in the AI Addendum Cover Page through conversation. This governs the use of AI tools and services in your commercial agreement, including data use, model training restrictions, and ownership of AI-generated outputs. Let's start -- what AI services are covered by this addendum? (e.g. 'AI-powered document analysis', 'Machine learning recommendation engine', 'Natural language processing chatbot')",
  pdfFilename: "ai-addendum.pdf",
};

File: backend/doc_configs/ai_addendum.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in an AI Addendum Cover Page.

## Fields you must collect

1. **Effective Date** -- when the addendum takes effect
   Ask: "When should this AI Addendum take effect? (e.g. 'today', 'January 1 2027', 'next Monday')"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15"), convert to YYYY-MM-DD format directly.

2. **AI Services** -- description of the AI services covered
   Ask: "What AI services are covered by this addendum? (e.g. 'AI-powered document analysis', 'Machine learning recommendation engine', 'Natural language processing chatbot')"

3. **Data Use Restrictions** -- restrictions on how customer data may be used with AI
   Ask: "What restrictions should apply to how customer data is used with AI? (e.g. 'No use of customer data for model training', 'Data may only be used for inference, not fine-tuning', 'All data must be anonymized before processing')"

4. **Governing Law** -- jurisdiction whose laws govern the addendum
   Ask: "Which state or jurisdiction's law should govern this addendum? (e.g. 'California', 'New York', 'Delaware')"

5. **Chosen Courts** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes? (e.g. 'San Francisco, California', 'New York County, New York')"

6. **Party 1 (Provider)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of AI Corp, jane@aicorp.com, signing today"

7. **Party 2 (Customer)** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the date"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    effectiveDate: Optional[str] = None
    aiServices: Optional[str] = None
    dataUseRestrictions: Optional[str] = None
    governingLaw: Optional[str] = None
    chosenCourts: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields
